best-practice-certificate-provisioning.md

```diff
@@ -0,0 +1,313 @@
+
+### Initial Certificate Provisioning
+
+#### EST Server Discovery
+
+On connection to the target environments network the EST Client SHOULD attempt to discover the location of the EST Server using [DNS-SD](#dns-sd-advertisement), if EST has not been disabled or the location of the EST configured manually.
+
+#### Get Root CA
+
+The EST Client SHOULD make a HTTPS request to the `/cacerts` endpoint of the EST Server for the latest Root CA of the current network. The EST Client SHOULD explicitly trust the EST Server manually configured or discovered using Unicast DNS and not perform authentication of the EST Servers TLS Certificate during the initial request to EST Server. If EST Server returns a HTTP 200 response, the EST Client should use the returned Root CA to authenticate all further communication with the EST Server and NMOS APIs.
+
+#### Generate Certificate Signing(CSR) Request
+
+The EST Client SHOULD create a CSR for each digital signature algorithm it supports with an appropriate Key Length. The CSR MUST contain a Common Name that is resolvable via DNS on the current domain and appropriate values for the other CSR fields. The manufacturer MAY allow the optional CSR fields to be configured by the operator.
+
+#### Certificate Request
+
+For each generated CSR the EST CLient SHOULD make a HTTPS request containing the CSR to the `/simpleenroll` endpoint of the EST Server. The EST Client SHOULD include the manufacturer installed TLS Client Certificate if present and valid during the TLS handshake with the EST Server. If the TLS certificate is no longer valid, the [Expired Manufacturer Issued TLS Client Certificate](#Expired-Manufacturer-Issued-TLS-Client-Certificate) workflow should be followed.
+
+If the EST Server returns a HTTP 200 response the certificate request was successful and the EST Client should use the returned TLS Certificate and its chain of trust for all further requests to its NMOS APIs.
+
+If the EST Server returns a HTTP 202 or HTTP 503 response, the request was successful, but the certificate has not been processed yet. The response SHOULD include a `Retry-After` header and the EST Client MUST not attempt re-sending the request before the defined time has expired. The EST Client SHOULD attempt resending the request 5 times before aborting the certificate request and attempting with an a alternative EST Server. [RFC 7030 Section 4.2.3](https://tools.ietf.org/html/rfc7030#section-4.2.3)
+
+If the EST Server returns any other HTTP response, the request has been unsuccessful, this could be caused by a malformed request, server side error or the EST Client not being authorised. The EST Client SHOULD restart the EST workflow with an alternative EST Server if present, else the EST Client SHOULD wait a backoff period 2 seconds plus jitter before retrying.
```

Comment:
Should the 2 seconds increase exponentially? Is this defined in an RFC we could reference instead?

Reply:
Not defined anywhere, thought 2 seconds was a sensible back off period, allowing the EST server to not be overloaded.

Reply:
Not seen an RFC that could be referenced.
Happy to say exponential backoff should be implemented instead, as is stated in IS-04, with not times specified.

Reply:
I'd suggest stating exponential backoff with a minimum starting period of 1 or 2 seconds. Given this is unlikely to be particularly time critical we could probably safely say higher if necessary.

Reply:
Added